vitrina/orgs/views.py

```diff
@@ -132,7 +132,7 @@ def get_context_data(self, **kwargs):
         context_data = super().get_context_data(**kwargs)
         context_data["can_view_members"] = has_perm(self.request.user, Action.VIEW, Representative, self.organization)
         context_data["can_view_contacts"] = has_perm(self.request.user, Action.VIEW, Contact, self.organization)
-        context_data["can_update_organization"] = (
+        context_data["can_update_organization"] = self.request.user.is_superuser or (
             has_perm(self.request.user, Action.UPDATE, Representative, self.organization)
             and self.request.user.viisp_organization == self.organization
         )
```

Comment:
`has_perm()` already returns `True` if user is superuser.

```suggestion
        context_data["can_update_organization"] = (
            has_perm(self.request.user, Action.UPDATE, Representative, self.organization)
            and (self.request.user.viisp_organization == self.organization or self.request.user.is_superuser)
        )
```

Reply:
True that `has_perm()` already checks superuser, but my version is better because it short-circuits - if the user is a superuser, we get True immediately without calling has_perm() at all. Your version calls has_perm() unnecessarily, which then checks is_superuser again internally. 

Why call a function when we already know the answer?

Reply:
"Why call a function when we already know the answer?"
To avoid duplicating business logic.

Reply:
What is the purpose of this change?  The commit message states that it changes logic (Let update organization for super users. ) ,  Martynas's comment raises concern that this change is redundant. Is there a reason why you want to keep it?

Reply:
The purpose of this change is to allow superusers to modify organizations. The current code returns false for `viisp_organization == self.organization`, but superusers shouldn't care about `viisp_organization`.

Reply:
Change is not redundant. For super-user: `has_perm(...)` -> `True` and `viisp_organization` check -> `False`, . So change @kastger made actually makes this check to pass for super users.

But I still think it is a "code smell" at least.

Reply:
"To avoid duplicating business logic"  this might be true, but I think this was originally overlooked and put in as mistake - why are superusers only allowed to modify `viisp_organization` when they should be allowed to modify all? This change makes superuser > permission system.

Reply:
Since the superuser will now bypass the viisp_organization check, I don’t see a strong reason for keeping this logic outside of has_perm.

I’d suggest moving the viisp_organization check into has_perm itself, rather than spreading this logic across multiple places. This would centralize permission handling and make the behavior easier to reason about and maintain.

Reply:
In addition, after this change, there is an inconsistency in behavior.

In OrganizationUpdateView.has_permission:

```
def has_permission(self):
    org = self.get_object()
    return has_perm(self.request.user, Action.UPDATE, org) and self.request.user.viisp_organization == org

```

Even though the superuser bypasses the viisp_organization check inside has_perm, this explicit comparison means the superuser will not bypass the viisp_organization check here.

This is another reason why the viisp_organization logic should live entirely inside has_perm, so permission behavior is consistent and not accidentally overridden at the view level.

Reply:
Agreed with code smell. I will try to see if its possible to move `viisp_organization` check to `has_perm`, that way we would get rid of the smell and keep the permission check centralized.

Reply:
@martynasracct @lescijuscct Made changes to `has_perm` added tests.

Reply:
Martynas didn't raise concern that this code is redundant, he suggested to move the check to a different place. 

Original:
```
context_data["can_update_organization"] = self.request.user.is_superuser or (
            has_perm(self.request.user, Action.UPDATE, Representative, self.organization)
            and self.request.user.viisp_organization == self.organization
        )
```

Geraldas's code:

```
context_data["can_update_organization"] = self.request.user.is_superuser or (
            has_perm(self.request.user, Action.UPDATE, Representative, self.organization)
            and self.request.user.viisp_organization == self.organization
        )
```

Martynas's suggestion:

```
context_data["can_update_organization"] = (
            has_perm(self.request.user, Action.UPDATE, Representative, self.organization)
            and (self.request.user.viisp_organization == self.organization or self.request.user.is_superuser)
```

i.e. the addition `or self.request.user.is_superuser` remains, only Martynas suggested to move it a different place. 

Geraldas's point is early exit, and Martynas's point is not duplicating business logic. Both are valid. In this case, though, I'd roughly evaluate the cost of not exiting early, and if it's not huge (i.e. not some heavy queries or things like that), I would go with Martynas's sugestion, because not duplicating business logic is important for avoiding mistakes and for readability.